src/bundle/Templating/Twig/UserPreferenceRuntime.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Bundle\AdminUi\Templating\Twig;
+
+use Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException;
+use Ibexa\Contracts\Core\Repository\Exceptions\UnauthorizedException;
+use Ibexa\Contracts\Core\Repository\UserPreferenceService;
+use Ibexa\Contracts\Core\Repository\Values\UserPreference\UserPreference;
+use Twig\Extension\RuntimeExtensionInterface;
+
+final class UserPreferenceRuntime implements RuntimeExtensionInterface
+{
+    private UserPreferenceService $userPreferenceService;
+
+    public function __construct(
+        UserPreferenceService $userPreferenceService
+    ) {
+        $this->userPreferenceService = $userPreferenceService;
+    }
+
+    public function getUserPreference(string $identifier): ?UserPreference
+    {
+        try {
+            return $this->userPreferenceService->getUserPreference($identifier);
```

Comment:
```suggestion
            return $this->userPreferenceService->getUserPreference($identifier)->value;
```

Reply:
I believe this feature will have more value in the ^ form

Reply:
> I believe this feature will have more value in the ^ form

I have doubts about that. For example, in https://github.com/ibexa/taxonomy/pull/253 we decided to return an `entry` object, instead of just a name. I see space for another function that will fetch all the preferences of the user. It will return the list of objects. I think that we should return the whole object, but I am open to discussion.

Reply:
We tend to return API objects in case someone wants to use other properties.

Reply:
What would Twig consumption look like in both cases?

Reply:
In case of checking if the preference is set you could use `{{ ibexa_get_user_preference('foo') is not null }}`. To fetch the value it will be {{ ibexa_get_user_preference('foo').value }}